### ISSUE TITLE
Move all the `Processor`s out of the main repo.

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4912]
We have a lot of `Processor`s:

- `KtFileCountProcessor`
- `PackageCountProcessor`
- `ClassCountProcessor`
- `FunctionCountProcessor`
- `PropertyCountProcessor`
- `ProjectComplexityProcessor`
- `ProjectCognitiveComplexityProcessor`
- `ProjectLLOCProcessor`
- `ProjectCLOCProcessor`
- `ProjectLOCProcessor`
- `ProjectSLOCProcessor`

And they are used to diplay in the html report and some console reports. Do we want to maintain this? It's true that it doesn't give any maintainance work but I'm wondering if anyone really use this kind of information. 

My proposal: Move all this code out of the main repo but keep it as a first-party plugin if someone wants to use it. Later we can decide what we want to do with that on `2.0`.

Why? Detekt is huge and we should focus on the things that detekt does the best. And I think that this is not one of those.

**Comment by schalkms** [comment]
I'm not in favor of scattering logic in multiple repositories.
It's true that those processors don't require much maintenance effort, though.

Furthermore, other rule packages, like for instance `detekt-rules-complexity`, depend on the `detekt-metrics` package. Thus `detekt-metrics` is required and not opt-in. Sure, one can refactor the package to only include necessary functionality. But what would one really gain by doing that?

**Comment by BraisGabin** [comment]
Reduce the number of processors and console reports dramatically from the default project. As an intermediate point we could keep them in the main repo but move them as a plugin. We could simplify the default configuration  from this:

```yml
processors:
  active: true
  exclude:
    - 'DetektProgressListener'
  # - 'KtFileCountProcessor'
  # - 'PackageCountProcessor'
  # - 'ClassCountProcessor'
  # - 'FunctionCountProcessor'
  # - 'PropertyCountProcessor'
  # - 'ProjectComplexityProcessor'
  # - 'ProjectCognitiveComplexityProcessor'
  # - 'ProjectLLOCProcessor'
  # - 'ProjectCLOCProcessor'
  # - 'ProjectLOCProcessor'
  # - 'ProjectSLOCProcessor'
  # - 'LicenseHeaderLoaderExtension'

console-reports:
  active: true
  exclude:
     - 'ProjectStatisticsReport'
     - 'ComplexityReport'
     - 'NotificationReport'
     - 'FindingsReport'
     - 'FileBasedFindingsReport'
  #  - 'LiteFindingsReport'
```

to this:

```yml
processors:
  active: true
  exclude:
    - 'DetektProgressListener'
  # - 'LicenseHeaderLoaderExtension'

console-reports:
  active: true
  exclude:
     - 'NotificationReport'
     - 'FindingsReport'
     - 'FileBasedFindingsReport'
  #  - 'LiteFindingsReport'
```
